whirlpool/appliance.py

```diff
@@ -35,20 +35,25 @@ def __init__(
         self._session: aiohttp.ClientSession = session
 
     def register_attr_callback(self, update_callback: Callable):
+        """Register Callback function."""
         self._attr_changed.append(update_callback)
         LOGGER.debug("Registered attr callback")
 
     def unregister_attr_callback(self, update_callback: Callable):
-        try:
-            self._attr_changed.remove(update_callback)
-            LOGGER.debug("Unregistered attr callback")
-        except ValueError:
-            LOGGER.error("Attr callback not found")
+        """Unregister callback function."""
+        if self._attr_changed:
```

Comment:
in what situations can this be None? It is inited in __init__ and is never set to None, so I believe we should not need this check

Reply:
You may be correct here - I put this test in the "wrong place" trying to stop the NoneType iteration error.